src/xclient_zmq_impl.cpp

```diff
@@ -0,0 +1,187 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include <iostream>
+
+#include "xeus-zmq/xauthentication.hpp"
+#include "xclient_zmq_impl.hpp"
+#include "xeus-zmq/xzmq_serializer.hpp"
+
+namespace xeus
+{
+
+    xclient_zmq_impl::xclient_zmq_impl(zmq::context_t& context,
+                                    const xeus::xconfiguration& config,
+                                    nl::json::error_handler_t eh)
+        : p_auth(make_xauthentication(config.m_signature_scheme, config.m_key))
+        , m_shell_client(context, config.m_transport, config.m_ip, config.m_shell_port)
+        , m_control_client(context, config.m_transport, config.m_ip, config.m_control_port)
+        , m_iopub_client(context, config)
```

Comment:
Maybe we can pass `transport`, `ip` and `port` here for `xiopub_client` too ?

Reply:
Yes that would make the code homogeneous, although this is nor really required.

Reply:
Sure, let's leave it like this for now.